akka-actor-tests/src/test/scala/akka/util/TokenBucketSpec.scala

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.util
+
+import akka.testkit.AkkaSpec
+
+import scala.util.Random
+
+class TokenBucketSpec extends AkkaSpec {
+
+  class TestBucket(_cap: Long, _period: Long) extends TokenBucket(_cap, _period) {
+    var currentTime: Long = 0L
+  }
+
+  "A Token Bucket" must {
```

Comment:
It helps because the person who is going to add or fix test cases do not have to derive, manually, the laws governing the implementation, but can rather read them based on the generators. :)